File: Magazin Aspnet/Data/Services/Impl/ProductServiceImpl.cs
namespace Magazin.Data.Services.Impl
{
    public class ProductServiceImpl : IProductService
    {
        private readonly ApplicationDbContext _context;
        public ProductServiceImpl(ApplicationDbContext context)
        {
            _context = context;
        }
        public void disableProduct(long id)
        {
            var product = _context.Products.FirstOrDefault(p => p.Id == id);
            if (product != null)
            {
                product.Active = false;
                _context.SaveChanges();
            }
        }

        public void enableProduct(long id)
        {
            var product = _context.Products.FirstOrDefault(p => p.Id == id);
            if (product != null)
            {
                product.Active = true;
                _context.SaveChanges();
            }
        }

        public List<Product> getAll()
        {
            return _context.Products.ToList();
        }

        public List<Product> getAllByCategory(string category)
        {
            return _context.Products.Where(p => p.Category.Equals(category)).ToList();
        }

        public Product getById(long id)
        {
            return _context.Products.FirstOrDefault(p => p.Id == id);
        }

        public Product getByProductLink(string productLink)
        {
            return _context.Products.FirstOrDefault(p => p.ProductLink.Equals(productLink));
        }

        public void saveProduct(Product product)
        {
            product.Active = true;
            _context.Products.Add(product);
            _context.SaveChanges();
        }

        public void updateProduct(Product product)
        {
            _context.Update(product);
            _context.SaveChanges();
        }
    }
}

File: Magazin Spring/src/main/java/com/shop/services/impl/ProductServiceImpl.java
package com.shop.services.impl;

import java.util.ArrayList;
import java.util.List;
import java.util.Optional;

import org.springframework.stereotype.Service;

import com.shop.repository.ProductRepository;
import com.shop.services.ProductService;
import com.shop.struct.Product;

@Service
public class ProductServiceImpl implements ProductService {

    private final ProductRepository productRepository;

    public ProductServiceImpl(ProductRepository productRepository) {
        this.productRepository = productRepository;
    }

    @Override
    public Optional<Product> getById(Long id) {
        return productRepository.findById(id);
    }

    @Override
    public Optional<Product> getByProductLink(String productLink) {
        return productRepository.findByproductlink(productLink);
    }

    @Override
    public List<Product> getAll() {
        List<Product> products = new ArrayList<>();
        productRepository.findAll().forEach(products::add);
        return products;
    }

    @Override
    public List<Product> getAllByCategory(String category) {
        return new ArrayList<>(productRepository.findAllByCategory(category));
    }

    @Override
    public void saveProduct(Product product) {
        product.setActive(true);
        productRepository.save(product);
    }

    @Override
    public void disableProduct(Long id) {
        Optional<Product> product = productRepository.findById(id);
        if (product.isPresent()) {
            product.get().setActive(false);
            productRepository.save(product.get());
        }
    }

    @Override
    public void enableProduct(Long id) {
        Optional<Product> product = productRepository.findById(id);
        if (product.isPresent()) {
            product.get().setActive(true);
            productRepository.save(product.get());
        }
    }
}
